pimd/pim_ifchannel.c

```diff
@@ -164,43 +166,54 @@ void pim_ifchannel_delete(struct pim_ifchannel *ch)
 	if (ch->sources)
 		list_delete(&ch->sources);
 
-	listnode_delete(ch->upstream->ifchannels, ch);
+	/* this can be called when the channel wasn't created yet */
+	if (ch->upstream && ch->upstream->ifchannels) {
+		listnode_delete(ch->upstream->ifchannels, ch);
+	}
```

Comment:
If the channel itself is not created yet, then why you are calling pim_ifchannel_delete() to delete the if channel.